File: platform/windows/AdGuard.FilterListManager/MarshalLogic/RulesCountByFilter.cs
namespace AdGuard.FilterListManager.MarshalLogic
{
    /// <summary>
    /// Rules count in this filter list. Simply a number of non-empty lines
    /// and does not start with a comment marker.
    /// </summary>
    public class RulesCountByFilter
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="RulesCountByFilter"/> class.
        /// </summary>
        /// <param name="filterId">The filter identifier.</param>
        /// <param name="rulesCount">The rules count.</param>
        public RulesCountByFilter(int filterId, int rulesCount)
        {
            FilterId = filterId;
            RulesCount = rulesCount;
        }

        /// <summary>
        /// Gets the filter identifier.
        /// </summary>
        public int FilterId { get; }

        /// <summary>
        /// Gets the rules count.
        /// </summary>
        public int RulesCount { get; }
    }
}

File: crates/filter-list-manager/src/manager/models/rules_count_by_filter.rs
use crate::FilterId;

/// List of rules count + filter id
pub struct RulesCountByFilter {
    /// Associated filter id.
    pub filter_id: FilterId,
    /// Rules count in this filter list. Simply a number of non-empty lines
    /// and does not start with a comment marker. See [`RulesListService::is_line_is_rule`](`crate::storage::services::rules_list_service::RulesListService::is_line_is_rule`).
    pub rules_count: i32,
}
